moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -111,22 +110,17 @@ const ComponentForm = ({
   );
 
   const phaseOptions = usePhaseOptions(optionsData);
-  const [
-    component,
-    phase,
-    completionDate,
-    subcomponents,
-    signal,
-  ] = watch([
+  const [component, phase, completionDate, subcomponents, signal] = watch([
     "component",
     "phase",
     "completionDate",
     "subcomponents",
     "signal",
   ]);
   const subphaseOptions = useSubphaseOptions(phase?.data.moped_subphases);
-  const internalTable = component?.data?.feature_layer?.internal_table;
-  const isSignalComponent = internalTable === "feature_signals";
+  const assetFeatureTable =
+    component?.data?.asset_feature_layer?.internal_table;
+  const isSignalComponent = assetFeatureTable === "feature_signals";
```

Comment:
I don't think it's big deal, but we still have some awkward hardcoding of component types and layer names in the editor. Maybe we can try to abstract this a bit when we look to add additional asset types (e.g. flashing beacons).